test/unittests/parser_test.cpp

```diff
@@ -40,23 +41,25 @@ bytes make_invalid_size_section(uint8_t id, size_t size, const bytes& content)
 
 TEST(parser, valtype)
 {
-    uint8_t b;
-    b = 0x7e;
-    EXPECT_EQ(std::get<0>(parse<ValType>(&b)), ValType::i64);
-    b = 0x7f;
-    EXPECT_EQ(std::get<0>(parse<ValType>(&b)), ValType::i32);
-    b = 0x7c;
-    EXPECT_THROW_MESSAGE(parse<ValType>(&b), parser_error, "unsupported valtype (floating point)");
-    b = 0x7d;
-    EXPECT_THROW_MESSAGE(parse<ValType>(&b), parser_error, "unsupported valtype (floating point)");
-    b = 0x7a;
-    EXPECT_THROW_MESSAGE(parse<ValType>(&b), parser_error, "invalid valtype 122");
+    std::array<uint8_t, 1> b{};
+    b[0] = 0x7e;
+    EXPECT_EQ(std::get<0>(parse<ValType>(b.begin(), b.end())), ValType::i64);
+    b[0] = 0x7f;
+    EXPECT_EQ(std::get<0>(parse<ValType>(b.begin(), b.end())), ValType::i32);
+    b[0] = 0x7c;
+    EXPECT_THROW_MESSAGE(
+        parse<ValType>(b.begin(), b.end()), parser_error, "unsupported valtype (floating point)");
+    b[0] = 0x7d;
+    EXPECT_THROW_MESSAGE(
+        parse<ValType>(b.begin(), b.end()), parser_error, "unsupported valtype (floating point)");
+    b[0] = 0x7a;
+    EXPECT_THROW_MESSAGE(parse<ValType>(b.begin(), b.end()), parser_error, "invalid valtype 122");
 }
 
 TEST(parser, valtype_vec)
 {
     const auto input = "037f7e7fcc"_bytes;
-    const auto [vec, pos] = parse_vec<ValType>(input.data());
+    const auto [vec, pos] = parse_vec<ValType>(input.data(), input.data() + input.size());
```

Comment:
Can't be `(input.begin(), input.end())` because iterators are not pointers there?

Reply:
Exactly. They are not pointers for `bytes`, but are for `bytes_view` (not guaranteed though).